docs/pages/principles/testing.md

```diff
@@ -0,0 +1,94 @@
+## Unit Tests
+
+### Advantages of unit testing:
+
+- Unit tests ensure that the code, as written, is correct, and executes
+  properly.
+- They communicate the intention of the creator of the code, how the code is
+  expected to behave, in its expected use-case.
+
+- Writing unit tests can reveal weakensses in our implementations, and lead us
+  to better design decisions:
+  - If the test requires excessive setup, the unit may be dependent on too many
+    external variables.
+  - If the test requires many assertions, the unit may be doing too many things
+    / have too many side-effects.
+  - If the unit is very difficult to test, it will likely be difficult to
+    understand and maintain. Refactoring code to make it easier to test often
+    leads us to write better code overall.
+
+### Guidelines for unit testing:
+
+- Unit tests live alongside the code they test, in a /tests folder.
+  - It looks like `unittest discover` cannot find TestCase classes within source
+    files.
+
+- Test files should be named `test_{{file under test}}.py`, so that stdlib
+  unittest can find them easily.
+
+- test\_.py files should match your source files (file-under-test) one-to-one,
+  and contain only tests for code in the file-file-under test.
```

Comment:
file-file-under? Also, `test_*.py` looks better, I think. I've seen people try to name a test file `test.py`. Doesn't end well. :)